Community report 2021 (#6409)
kubernetes/community: sig-windows/annual-report-2021.md

   - Alpha
     - [2258 - Node service log viewer](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/2258-node-service-log-viewer/README.md)
   - Pre-alpha
     - [2578 - Windows Conformance](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/2578-windows-conformance/kep.yaml)

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)
   - csi-proxy and storage: this seems like an underserved area for windows
2. What metrics/community health stats does your group care about and/or measure?
   - n/a
3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
   to activities or programs that provide useful context or allow easy participation?
   - yes
4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
   does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?

>    - yes

Reviewer: The contributing.md looks great for new contributors to get started. I'm not seeing training/requirements/processes for existing contributors there... am I missing it?

5. Does the group have contributors from multiple companies/affiliations?
   - yes

6. Are there ways end users/companies can contribute that they currently are not?
   If one of those ways is more full time support, what would they work on and why?
   - testing hostProcess implementations on several windows apps

>    - improving our dev tools enviornment to grow the commnunity

Reviewer: ```suggestion
   - improving our dev tools environment to grow the community
```

   - hardening the CSI proxy and CSI support ecosystem

>    - performance testing Kuberentes on Windows extensively and publishing results

Reviewer: ```suggestion
   - performance testing Kubernetes on Windows extensively and publishing results
```

> 

Reviewer: are there linkable details or points of contact for anyone wanting to help with these?

## Membership
- Primary slack channel member count: 1507
- Primary mailing list member count: 188
- Primary meeting attendee count (estimated, if needed): 10
- Primary meeting participant count (estimated, if needed): 10
- Unique reviewers for SIG-owned packages: 6
- Unique approvers for SIG-owned packages: 4

Include any other ways you measure group membership

## Subprojects

> Subprojects for sig-windows are inactive now and all major work happens in the sig-windows channel.

Reviewer: should the subprojects in sigs.yaml be updated?

## Current initiatives

1. What work did the SIG do this year that should be highlighted?

>    - Lead the transition to hostProcess containers.

Reviewer 2: ```suggestion
   - Implemented hostProcess container support in Kubernetes (now in beta) and pomoted adoption in multiple open source communities
     - https://github.com/kubernetes-sigs/sig-windows-tools/tree/master/hostprocess - for examples of running flannel, calico, csi-proxy, kube-proxy as hostPorcess containers
     - https://github.com/weaveworks/kured/pull/460  - KuReD Windows support
     - https://github.com/prometheus-community/windows_exporter/pull/864/files - node exporter support
```

   - Defined the `kubectl node logs` command interface.
   - Made the developer UX for windows transparent with sig-windows-dev-tools.

>    - Defined windows operational readiness standards.

Reviewer 3: Should the PodOS field work be added here?

   
2. What initiatives are you working on that aren't being tracked in KEPs?

>    - Migration of the windows kube-proxy to KPNG (led by amim).

Reviewer 2: ```suggestion
   - Migration of the windows kube-proxy to KPNG.
```

   - Migration of testgrid reporting jobs from aks-engine to cluster-api/cluster-api-provder-azure.
   - Dockershim removal / validation for Windows nodes.

3. KEP work in 2021 (1.x, 1.y, 1.z):
   - Stable
     - (1.22) [1122 - windows-csi-support](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/1122-windows-csi-support/README.md)
   - Beta
     - (1.23) [1981 - Windows Privileged Container Support](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/1981-windows-privileged-container-support/README.md)
     - (1.23) https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/2802-identify-windows-pods-apiserver-admission/kep.yaml
   - Alpha
     - (1.22) [1981 - Windows Privileged Container Support](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/1981-windows-privileged-container-support/README.md)
     - (1.23) [2802 -Identify Windows pods at API admission level authoritatively](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/2802-identify-windows-pods-apiserver-admission/README.md)
   - Pre-alpha (Targeting 1.24)

>      - [2578 - Windows Conformance](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/2578-windows-conformance/kep.yaml)

Reviewer 3: is this the right title?  I thought we went with "Operational Readiness"

Reviewer 2: Looks like some references got updated (like the k/ehancements issue https://github.com/kubernetes/enhancements/issues/2578) but others (like the key.yaml - https://github.com/kubernetes/enhancements/blob/18b4a6fd320bbbd4faed9a2cf531b5fc358df263/keps/sig-windows/2578-windows-conformance/kep.yaml#L1) did not.

@knabben can you clean up the Windows-conformance references when you get a chance?

Author: fixed it to say op-readiness

## Current initiatives

1. What work did the SIG do this year that should be highlighted?
   - Implemented hostProcess container support in Kubernetes (now in beta) and pomoted adoption in multiple open source communities

>      - https://github.com/kubernetes-sigs/sig-windows-tools/tree/master/hostprocess - for examples of running flannel, calico, csi-proxy, kube-proxy as hostPorcess containers

Reviewer: ```suggestion
     - https://github.com/kubernetes-sigs/sig-windows-tools/tree/master/hostprocess - for examples of running flannel, calico, csi-proxy, kube-proxy as hostProcess containers
```

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)
   - csi-proxy and storage: this seems like an underserved area for windows https://github.com/kubernetes-csi/csi-proxy (meeting info is there).
2. What metrics/community health stats does your group care about and/or measure?
   - stars for ksigs/windows specific repos
     - sig-windows-dev-tools   
       - https://github.com/kubernetes-sigs/sig-windows-dev-tools -> up to 46, represents interest
     - sig-windows-tools
       - represents people trying to install windows on k8s nodes
     - windows-gmsa
       - represents enterprises integrating windows pods into GMSA
3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
   to activities or programs that provide useful context or allow easy participation?
   - yes
4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
   does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?

>    - yes

Reviewer: still not seeing this... if this isn't there yet, describe the gap and link to an issue tracking resolving it?

Reviewer 2: I recently updated the contributing guide with #6433

https://github.com/kubernetes/community/blob/master/sig-windows/CONTRIBUTING.md#creating-a-pr mentions to follow the general contributing guide but add `/sig windows` and trigger additional e2e PRs

PTAL

Reviewer: thanks

5. Does the group have contributors from multiple companies/affiliations?
   - yes

6. Are there ways end users/companies can contribute that they currently are not?
   If one of those ways is more full time support, what would they work on and why?
   - testing hostProcess implementations on several windows apps
   - improving our dev tools environment to grow the community
   - hardening the CSI proxy and CSI support ecosystem

>    - performance testing Kubernetes on Windows extensively and publishing results

Reviewer: where should someone who sees these and wants to help go? are there topics or tracking issues for these?

Author: added specific target of blog post

>    - performance testing Kubernetes on Windows extensively and publishing results

Author: ```suggestion
   - performance testing Kubernetes on Windows extensively and publishing results in cncf blog posts
```

## Membership
- Primary slack channel member count: 1507
- Primary mailing list member count: 188
- Primary meeting attendee count (estimated, if needed): 10
- Primary meeting participant count (estimated, if needed): 10
- Unique reviewers for SIG-owned packages: 6
- Unique approvers for SIG-owned packages: 4

Include any other ways you measure group membership

## Subprojects

- windows csi-proxy subproject is active and healthy https://github.com/kubernetes-csi/csi-proxy
    - meetings going well
    - new optimization issue came up recently community engaged on it across companies vmware,rancher

## Working groups

> Same as above.

Reviewer: if sig-windows doesn't have any working groups (it doesn't look like they do), can mark this as n/a

Author: fixed

> Same as above.

Author: ```suggestion
n/a
```

> ## Operational

Reviewer: can iterate on these in a follow-up in the next week or so, but the goal is to resolve them as part of closing https://github.com/kubernetes/community/issues/6386 to make sure these are accurate at least once a year
